wazo_call_logd/purger.py

```diff
@@ -54,17 +54,24 @@ def _extract_days_to_keep(tenant_days, default_days, purge_db_days, default):
     return default_days
 
 
+def _get_tenants_uuids(session, tenant_uuid):
+    if tenant_uuid:
+        return [tenant_uuid]
+    else:
+        return [tenant_uuid[0] for tenant_uuid in session.query(Tenant.uuid).all()]
+
+
 class CallLogsPurger:
-    def purge(self, days_to_keep, session):
+    def purge(self, days_to_keep, session, tenant_uuid=None):
         config = session.query(Config).first()
         if not config:
             raise Exception('No default config found')
         default_days = config.retention_cdr_days
 
         retentions = {r.tenant_uuid: r for r in session.query(Retention).all()}
-        tenants = session.query(Tenant).all()
-        for tenant in tenants:
-            retention = retentions.get(tenant.uuid)
+        tenants_uuid = _get_tenants_uuids(session, tenant_uuid)
+        for tenant_uuid in tenants_uuid:
```

Comment:
I find the interface/separation of concern around `_get_tenants_uuids` a bit weird. I understand the desire to avoid repetition, but I think logical decomposition and clarity is worth a bit of redundancy.

I feel the calling context should know and decide whether it has to use this function to make a database call. 
 _get_tenants_uuids can then just take just the `session` parameter and only be called if there's no `tenant_uuid` to use. That way there's no ambiguity about why you're calling the function and what its effect will be. 
 
 ```suggestion
        if not tenant_uuid
            tenants_uuid = _get_tenants_uuids(session)
        else:
            tenants_uuid = [tenant_uuid]
            
        for tenant_uuid in tenants_uuid:
```
That way `_get_tenants_uuids` has a clear purpose and scope, and is only called if needed.

Reply:
I want to avoid code duplication. The `_get_tenants_uuids` is pretty **short** and **clear** to read. So I would like to keep the `if not tenant_uuid ... ` in this function. 
Let me know if this is a big issue for you.